pkg/kubernetes/scripts/virtual-machines/utils.jsx

```diff
@@ -46,3 +46,29 @@ export function mouseClick(fun) {
         return fun(event);
     };
 }
+
+/**
+ * Returns sub-property value of given path
+ *
+ * Example:
+ *   getValueOrDefault(myObj, ['x', 'y'], defaultVal) will return myObj.x.y or defaultVal if path does not exist
+ *
+ */
+export function getValueOrDefault(obj, attrPath, defaultValue) {
```

Comment:
I believe Javascript provides better ways, e.g.

```js
function safeCall(accessor, defaultValue) {
  try {
    return accessor()
  } catch (error) {
    if (!(error instanceof TypeError)) {
      throw error
    }
  }
  return defaultValue
}

const foo = {}
safeCall(() => foo.bar.baz, 'there was nothing there')  // returns  'there was nothing there'
safeCall(() => foo.bar.baz)                             // returns  undefined
```

Such approach would allow us to keep ability to use
* grep
* IDE refactoring tools
* auto completion
* type annotations

Reply:
Thanks for posting that, this is better.